DSCResources/MSFT_xWebsite/MSFT_xWebsite.psm1

```diff
@@ -576,67 +596,374 @@ function ConvertTo-CimBinding
-                    CertificateStoreName  = [String]$Binding.CertificateStoreName
-                    SslFlags              = [String][Int32]$Binding.SslFlags
-                }
-            }
-            else
+            if ($Binding.Protocol -in @('http', 'https'))
             {
                 if ($Binding.bindingInformation -match '^\[(.*?)\]\:(.*?)\:(.*?)$')
                 {
                     $IPAddress = $Matches[1]
-                    $Port = $Matches[2]
-                    $HostName = $Matches[3]
+                    $Port      = $Matches[2]
+                    $HostName  = $Matches[3]
                 }
                 else
                 {
                     $IPAddress, $Port, $HostName = $Binding.bindingInformation -split '\:'
                 }
 
-                $CimProperties = @{
-                    Port                  = [UInt16]$Port
-                    Protocol              = [String]$Binding.Protocol
-                    IPAddress             = [String]$(if ($IPAddress) {$IPAddress} else {'*'})
-                    HostName              = [String]$HostName
-                    CertificateThumbprint = [String]$Binding.certificateHash
-                    CertificateStoreName  = [String]$Binding.certificateStoreName
-                    SslFlags              = [String][Int32]$Binding.sslFlags
+                if (-not $IPAddress)
+                {
+                    $IPAddress = '*'
                 }
+
+                $CimProperties.Add('IPAddress', [String]$IPAddress)
+                $CimProperties.Add('Port',      [UInt16]$Port)
+                $CimProperties.Add('HostName',  [String]$HostName)
+            }
+            else
+            {
+                $CimProperties.Add('IPAddress', [String]::Empty)
+                $CimProperties.Add('Port',      [UInt16]::MinValue)
+                $CimProperties.Add('HostName',  [String]::Empty)
             }
 
+            if ([Environment]::OSVersion.Version -ge '6.2')
+            {
+                $CimProperties.Add('SslFlags', [String]$Binding.sslFlags)
+            }
+
+            $CimProperties.Add('CertificateThumbprint', [String]$Binding.certificateHash)
+            $CimProperties.Add('CertificateStoreName',  [String]$Binding.certificateStoreName)
+
             New-CimInstance -ClassName $CimClassName -Namespace $CimNamespace -Property $CimProperties -ClientOnly
 
         }
     }
 }
 
+function ConvertTo-WebBinding
+{
+    <#
+    .SYNOPSIS
+        Converts instances of the MSFT_xWebBindingInformation CIM class to the IIS <binding> element representation.
+    .LINK
+        https://www.iis.net/configreference/system.applicationhost/sites/site/bindings/binding
+    #>
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true, ValueFromPipeline = $true)]
+        [AllowEmptyCollection()]
+        [AllowNull()]
+        [Object[]]
+        $InputObject
+    )
+    process
+    {
+        $InputObject |
```

Comment:
Personally I find $InputObject | Foreach-Object a bit harder to read.  And you end up just binding $_ to $Binding.

Could we just change this to
````powershell
process
{
    foreach ($binding in $InputObject)
   {
        # Code here
    }
}
````